src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -680,13 +680,13 @@ private void builderQueryParamEncoded(UriComponentsBuilder builder, String name,
                     unEncodedAsUsername += "@" + domain + ".commcarehq.org";
                 }
                 builderQueryParamEncoded(builder, "as", unEncodedAsUsername);
+            } else if (getHqAuth() == null && username != null) {
+                // HQ requesting to force a sync for a user
+                builderQueryParamEncoded(builder, "as", username);
```

Comment:
The value of the `for` parameter observed in while investigating the SMS forms issue was not a valid couch username (it did not contain `@{domain}.commcarehq.org`. Should this use logic similar to
https://github.com/dimagi/formplayer/blob/c2f08a60cbbf0ea5e76b37b578ed8077fb8def12/src/main/java/org/commcare/formplayer/services/RestoreFactory.java#L678-L681

Reply:
no since `username` is always the full username, it's only `asUsername` that can be just a mobile workers username without the suffix

Reply:
> `username` is always the full username

Hmm, that doesn't match up with what I observed when I was troubleshooting the issue yesterday. `for` and `as` were different (see *restore* breadcrumb in [this sentry error](https://sentry.io/organizations/dimagi/issues/2177035085/)). Anonymized: `Restoring from URL https://www.commcarehq.org/a/{domain}/phone/restore/?version=2.0&since=...&device_id=...&as={username}%40{domain}.commcarehq.org&for={username}`

Reply:
Sorry, I should have been more clear. This branch of the if block is only ever triggered by the 'prime formplayer restores'  feature which always sends the full username for the 'username' and the abbreviated username for 'as user'. I don't know why SMS forms use the abbreviated username for `username`.

I guess we could do a check here as well to be more defensive.